lib/src/tokenizer.cc

```diff
@@ -0,0 +1,555 @@
+            }
+            result += c;
+            c = next_char_raw();
+        }
+
+        // The last char we looked at wasn't part of the number, put it back
+        put_back(c);
+
+        try {
+            if (contained_decimal_or_E) {
+                return make_shared<value>(unique_ptr<config_double>(
+                        new config_double(_line_origin, boost::lexical_cast<double>(result), result)));
+            } else {
+                return make_shared<value>(unique_ptr<config_long>(
+                        new config_long(_line_origin, boost::lexical_cast<long>(result), result)));
+            }
+        } catch (boost::bad_lexical_cast const& ex) {
+            // not a number after all, see if it's an unquoted string
+            for (char character : result) {
+                if (not_in_unquoted_text.find(character) != string::npos) {
+                    throw config_exception("Line " + std::to_string(_line_number) + ": Reserved character '"
+                                            + character + "' not allowed outside quotes");
+                }
+            }
+            // no disallowed chars, so we decide this was a string and not a number
+            return make_shared<unquoted_text>(_line_origin, result);
+        }
+    }
+
+    void token_iterator::pull_escape_sequence(string& parsed, string& original) {
+        char escaped = next_char_raw();
+        if (escaped == -1) {
+            throw config_exception("End of input but backslash in string had nothing after it");
+        }
+
+        // This is needed so we return the unescaped escape characters back out when rendering
+        // the token
+        original += "\\";
+        original += escaped;
+
+        switch (escaped) {
+            case '"':
+                parsed += '"';
+                break;
+            case '\\':
+                parsed += '\\';
+                break;
+            case '/':
+                parsed += '/';
+                break;
+            case 'b':
+                parsed += '\b';
+                break;
+            case 'f':
+                parsed += '\f';
+                break;
+            case 'n':
+                parsed += '\n';
+                break;
+            case 'r':
+                parsed += '\r';
+                break;
+            case 't':
+                parsed += '\t';
+                break;
+            case 'u': {
+                char utf[5] = {};
+                for (int i = 0; i < 4; i++) {
+                    int c = next_char_raw();
+                    if (c == -1) {
+                        throw config_exception("End of input but expecting 4 hex digits for \\uXXXX escape");
+                    }
+                    utf[i] = c;
+                }
+                original += string(utf);
+                short character;
+                sscanf(utf, "%hx", &character);
+                wchar_t buffer[] { static_cast<wchar_t>(character), '\0'};
```

Comment:
Try converting buffer to a wstring with a length of 1. boost::nowide::narrow is probably doing strlen, which will see a length of 0 because we're dealing with the null byte.